[WPE][GTK] Share public headers

mcatanzaro opened the pull request:
#### 5916de2c19fe379f7af03ab56ef062540d434b07
<pre>
[WPE][GTK] Share public headers
<a href="https://bugs.webkit.org/show_bug.cgi?id=243809">https://bugs.webkit.org/show_bug.cgi?id=243809</a>

Reviewed by NOBODY (OOPS!).

We&apos;re soon going to have four public APIs:

 * GTK 3
 * GTK 4
 * WPE 1.0/1.1
 * WPE 2.0

To avoid having four different copies of all the public headers, let&apos;s
generate the public headers from a shared template. We&apos;ll start by
unifying the existing WPE and GTK headers without making any API
changes. Using a customized version of unifdef, we can use
PLATFORM(GTK) and PLATFORM(WPE) guards in the header templates in order
to hide platform-specific code from the generated headers. Be careful
around comments, though: the guards do not work inside comments, so if
any part of a doc comment needs to differ, the entire doc comment will
need to be duplicated.

Also note that I don&apos;t touch the DOM API. GTK 3 will just have to be
different from the other APIs.

Thanks to Adrian for proposing this solution.

This commit additionally removes the prohibition on tab characters under
Source/ThirdParty, since there&apos;s not really any great reason to care
about whitespace in third-party code, and we don&apos;t use Subversion
anymore so concerns about the Subversion allow-tabs property no longer
apply.

I also got rid of WebKitForwardDeclarations.h. Turns out it was only
needed to break an include cycle between GTK&apos;s WebKitWebView.h and
WebKitPrintOperation.h, but that can be solved easily enough without it.

I also added WebKitVersion.h to WPE&apos;s web process API, to match GTK.

* Source/ThirdParty/unifdef/CMakeLists.txt: Added.
* Source/ThirdParty/unifdef/README.webkit: Added.
* Source/ThirdParty/unifdef/unifdef.c: Added.
(processinout):
(cleantemp):
(version):
(synopsis):
(usage):
(help):
(Eelif):
(Eelse):
(Eendif):
(Eeof):
(Eioccc):
(print):
(drop):
(Strue):
(Sfalse):
(Selse):
(Pelif):
(Pelse):
(Pendif):
(Dfalse):
(Delif):
(Delse):
(Dendif):
(Fdrop):
(Fpass):
(Ftrue):
(Ffalse):
(Oiffy):
(Oif):
(Oelif):
(Idrop):
(Itrue):
(Ifalse):
(Mpass):
(Mtrue):
(Melif):
(Melse):
(ignoreon):
(keywordedit):
(nest):
(unnest):
(state):
(done):
(flushline):
(hashline):
(closeio):
(process):
(parseline):
(op_strict):
(op_lt):
(op_gt):
(op_le):
(op_ge):
(op_eq):
(op_ne):
(op_or):
(op_and):
(op_blsh):
(op_brsh):
(op_add):
(op_sub):
(op_mul):
(op_div):
(op_mod):
(op_bor):
(op_bxor):
(op_band):
(eval_unary):
(eval_table):
(ifeval):
(skiphash):
(skipline):
(skipcomment):
(skipargs):
(skipsym):
(getsym):
(matchsym):
(findsym):
(indirectsym):
(addsym1):
(addsym2):
(debugsym):
(defundefile):
(defundef):
(astrcat):
(xstrdup):
(debug):
(error):
* Source/ThirdParty/unifdef/unifdef.h: Added.
(mktempmode):
* Source/ThirdParty/unifdef/version.h: Added.
* Source/WebKit/GLib.cmake: Added.
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/UIProcess/API/glib/WebKitApplicationInfo.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitApplicationInfo.h.
* Source/WebKit/UIProcess/API/glib/WebKitAuthenticationRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitAuthenticationRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitAutocleanups.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitAutocleanups.h.
* Source/WebKit/UIProcess/API/glib/WebKitAutomationSession.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitAutomationSession.h.
* Source/WebKit/UIProcess/API/glib/WebKitBackForwardList.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitBackForwardList.h.
* Source/WebKit/UIProcess/API/glib/WebKitBackForwardListItem.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitBackForwardListItem.h.
* Source/WebKit/UIProcess/API/glib/WebKitContextMenu.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitContextMenu.h.
* Source/WebKit/UIProcess/API/glib/WebKitContextMenuActions.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitContextMenuActions.h.
* Source/WebKit/UIProcess/API/glib/WebKitContextMenuItem.h.in: Renamed from Source/WebKit/UIProcess/API/gtk3/WebKitContextMenuItem.h.
* Source/WebKit/UIProcess/API/glib/WebKitCookieManager.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitCookieManager.h.
* Source/WebKit/UIProcess/API/glib/WebKitCredential.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitCredential.h.
* Source/WebKit/UIProcess/API/glib/WebKitDefines.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitDefines.h.
* Source/WebKit/UIProcess/API/glib/WebKitDeviceInfoPermissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitDeviceInfoPermissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitDownload.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitDownload.h.
* Source/WebKit/UIProcess/API/glib/WebKitEditingCommands.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitEditingCommands.h.
* Source/WebKit/UIProcess/API/glib/WebKitEditorState.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitEditorState.h.
* Source/WebKit/UIProcess/API/glib/WebKitError.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitError.h.
* Source/WebKit/UIProcess/API/glib/WebKitFaviconDatabase.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitFaviconDatabase.h.
* Source/WebKit/UIProcess/API/glib/WebKitFileChooserRequest.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitFileChooserRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitFindController.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitFindController.h.
* Source/WebKit/UIProcess/API/glib/WebKitFormSubmissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitFormSubmissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitGeolocationManager.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitGeolocationManager.h.
* Source/WebKit/UIProcess/API/glib/WebKitGeolocationPermissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitGeolocationPermissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitHitTestResult.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitHitTestResult.h.
* Source/WebKit/UIProcess/API/glib/WebKitInputMethodContext.h.in: Renamed from Source/WebKit/UIProcess/API/gtk3/WebKitInputMethodContext.h.
* Source/WebKit/UIProcess/API/glib/WebKitInstallMissingMediaPluginsPermissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitInstallMissingMediaPluginsPermissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitJavascriptResult.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitJavascriptResult.h.
* Source/WebKit/UIProcess/API/glib/WebKitMediaKeySystemPermissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitMediaKeySystemPermissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitMemoryPressureSettings.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitMemoryPressureSettings.h.
* Source/WebKit/UIProcess/API/glib/WebKitMimeInfo.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitMimeInfo.h.
* Source/WebKit/UIProcess/API/glib/WebKitNavigationAction.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitNavigationAction.h.
* Source/WebKit/UIProcess/API/glib/WebKitNavigationPolicyDecision.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitNavigationPolicyDecision.h.
* Source/WebKit/UIProcess/API/glib/WebKitNetworkProxySettings.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitNetworkProxySettings.h.
* Source/WebKit/UIProcess/API/glib/WebKitNotification.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitNotification.h.
* Source/WebKit/UIProcess/API/glib/WebKitNotificationPermissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitNotificationPermissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitOptionMenu.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitOptionMenu.h.
* Source/WebKit/UIProcess/API/glib/WebKitOptionMenuItem.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitOptionMenuItem.h.
* Source/WebKit/UIProcess/API/glib/WebKitPermissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitPermissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitPlugin.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitPlugin.h.
* Source/WebKit/UIProcess/API/glib/WebKitPolicyDecision.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitPolicyDecision.h.
* Source/WebKit/UIProcess/API/glib/WebKitResponsePolicyDecision.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitResponsePolicyDecision.h.
* Source/WebKit/UIProcess/API/glib/WebKitScriptDialog.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitScriptDialog.h.
* Source/WebKit/UIProcess/API/glib/WebKitSecurityManager.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitSecurityManager.h.
* Source/WebKit/UIProcess/API/glib/WebKitSecurityOrigin.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitSecurityOrigin.h.
* Source/WebKit/UIProcess/API/glib/WebKitSettings.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitSettings.h.
* Source/WebKit/UIProcess/API/glib/WebKitURIRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitURIRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitURIResponse.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitURIResponse.h.
* Source/WebKit/UIProcess/API/glib/WebKitURISchemeRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitURISchemeRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitURISchemeResponse.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitURISchemeResponse.h.
* Source/WebKit/UIProcess/API/glib/WebKitURIUtilities.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitURIUtilities.h.
* Source/WebKit/UIProcess/API/glib/WebKitUserContent.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitUserContent.h.
* Source/WebKit/UIProcess/API/glib/WebKitUserContentFilterStore.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitUserContentFilterStore.h.
* Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitUserContentManager.h.
* Source/WebKit/UIProcess/API/glib/WebKitUserMediaPermissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitUserMediaPermissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitUserMessage.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitUserMessage.h.
* Source/WebKit/UIProcess/API/glib/WebKitVersion.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitVersion.h.in.
* Source/WebKit/UIProcess/API/glib/WebKitWebContext.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitWebContext.h.
* Source/WebKit/UIProcess/API/glib/WebKitWebResource.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitWebResource.h.
* Source/WebKit/UIProcess/API/glib/WebKitWebView.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitWebView.h.
* Source/WebKit/UIProcess/API/glib/WebKitWebViewSessionState.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitWebViewSessionState.h.
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteData.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitWebsiteData.h.
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataAccessPermissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitWebsiteDataAccessPermissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitWebsiteDataManager.h.
* Source/WebKit/UIProcess/API/glib/WebKitWebsitePolicies.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitWebsitePolicies.h.
* Source/WebKit/UIProcess/API/glib/WebKitWindowProperties.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitWindowProperties.h.
* Source/WebKit/UIProcess/API/gtk/WebKitApplicationInfo.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitAuthenticationRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitAutomationSession.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitBackForwardList.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitBackForwardListItem.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitContextMenu.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitDefines.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitDeviceInfoPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitDownload.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitEditingCommands.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitEditorState.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitFindController.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitForwardDeclarations.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitGeolocationManager.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitGeolocationPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitHitTestResult.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitInstallMissingMediaPluginsPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitMediaKeySystemPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitMemoryPressureSettings.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitMimeInfo.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitNavigationAction.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitNetworkProxySettings.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitNotification.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitNotificationPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitOptionMenu.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitOptionMenuItem.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitPlugin.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitPolicyDecision.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.h:
* Source/WebKit/UIProcess/API/gtk/WebKitResponsePolicyDecision.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitScriptDialog.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitSecurityManager.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitSecurityOrigin.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitURIRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitURIResponse.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitURISchemeRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitURISchemeResponse.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitURIUtilities.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitUserContent.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitUserContentFilterStore.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitUserContentManager.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitUserMediaPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitUserMessage.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitVersion.h.in: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitWebResource.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.h.in: Renamed from Source/WebKit/UIProcess/API/gtk3/WebKitWebViewBase.h.
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewSessionState.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitWebsiteData.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitWebsiteDataAccessPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitWebsiteDataManager.h: Removed.
* Source/WebKit/UIProcess/API/gtk4/WebKitContextMenuItem.h: Removed.
* Source/WebKit/UIProcess/API/gtk4/WebKitInputMethodContext.h: Removed.
* Source/WebKit/UIProcess/API/gtk4/WebKitWebViewBase.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitAutocleanups.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitContextMenuActions.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitContextMenuItem.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitCookieManager.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitCredential.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitError.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitFaviconDatabase.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitFileChooserRequest.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitFormSubmissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitInputMethodContext.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitJavascriptResult.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitNavigationPolicyDecision.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitSettings.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitWebContext.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitWebView.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitWebsitePolicies.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitWindowProperties.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitConsoleMessage.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitConsoleMessage.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitFrame.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitFrame.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitScriptWorld.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitScriptWorld.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebEditor.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebEditor.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebExtension.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebExtension.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebExtensionAutocleanups.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebExtensionAutocleanups.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebHitTestResult.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebHitTestResult.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebPage.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebPage.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/webkit-web-extension.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/gtk/webkit-web-extension.h.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitConsoleMessage.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitScriptWorld.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebEditor.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebExtension.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebExtensionAutocleanups.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebHitTestResult.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitFrame.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebPage.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/webkit-web-extension.h: Removed.
* Tools/Scripts/webkitpy/style/checker.py:
</pre>

carlosgcampos commented on Source/WebKit/PlatformGTK.cmake:
```diff
@@ -544,6 +551,34 @@ if (ENABLE_MEDIA_STREAM)
+
+    list(APPEND WebKit2WebExtension_INSTALLED_HEADERS "${WebKit2Gtk_DERIVED_SOURCES_DIR}/webkit/${HEADER}")
+endforeach ()
+list(APPEND WebKit_SOURCES ${WebKit2WebExtension_INSTALLED_HEADERS})
```
Could this be converted into a macro to avoid the code duplication?

mcatanzaro replied:
Probably: it really needs to be, because otherwise it will become a big mess as soon as we introduce any more build configurations. I was planning to attempt that in a follow-up patch, but I'll try now.

mcatanzaro replied:
Fixed, now uses a macro.

carlosgcampos commented on Source/WebKit/UIProcess/API/glib/WebKitApplicationInfo.h.in:
```diff
@@ -17,15 +17,26 @@
+#if !defined(__WEBKIT_H_INSIDE__) && !defined(WEBKIT2_COMPILATION)
+#error "Only <wpe/webkit.h> can be included directly."
+#endif
+#endif
```
Now that we are going to generate the headers, I wonder if we could simplify them even more. For example this could be something like `@PLATFORM_SINGLE_HEADER_CHECK@`, or something like that, that is substituted by the platform check. We would probably need a script that process the header and calls unifdef as the last step.

mcatanzaro replied:
Good idea, especially because this is going to become even more complicated if I succeed in dropping the `2` from the GTK API version, in which case we'll have three conditions here.

We *might* be able to avoid a new script if we define `PLATFORM_SINGLE_HEADER_CHECK` in a CMake variable and use configure_file().

Can we do this in the future rather than now, though? If it turns out to be very simple then I might submit a follow-up imminently, but if not, it's not important enough to block on IMO.

Also, keep in mind the required checks are different for UI process vs. InjectedBundle, and files that are shared will need yet another variant to allow both.

carlosgcampos replied:
Sure, we can improve it in follow up patches.

mcatanzaro replied:
Agreed to do this in a follow-up patch.

mcatanzaro replied:
Wound up doing it in this PR anyway.

carlosgcampos commented on Source/WebKit/UIProcess/API/glib/WebKitApplicationInfo.h.in:
```diff
@@ -17,15 +17,26 @@
+#elif PLATFORM(WPE)
+#include <wpe/WebKitDefines.h>
+#endif
 
```
And same here, we could convert the includes automatically, converting webkit2/ to wpe/ for wpe. Or using something like `#include <@PLATFORM_INCLUDE_PREFIX@/Foo.h>`  or if that's to ugly we could simply use `#include "Foo.h"` and convert all includes not using `<>` to `<prefix/Foo.h>`.

mcatanzaro replied:
Hm, if we do `#include <@PLATFORM_INCLUDE_PREFIX@/Foo.h>` then we might be able to use configure_file(). Otherwise, we'd definitely need a script.

mcatanzaro replied:
Agreed to do this in a follow-up patch.

mcatanzaro replied:
Wound up doing it in this PR anyway.

carlosgcampos commented on Source/WebKit/UIProcess/API/glib/WebKitAuthenticationRequest.h.in:
```diff
@@ -17,17 +17,29 @@
 #define WebKitAuthenticationRequest_h
 
+#if PLATFORM(GTK)
 #include <gtk/gtk.h>
```
And unifdef would still be needed for these cases where we really need a platform specific include.

carlosgcampos reviewed (commented):
I like this.

carlosgcampos reviewed (approved):
👌️

mcatanzaro committed:
[WPE][GTK] Share public headers
https://bugs.webkit.org/show_bug.cgi?id=243809

Reviewed by Carlos Garcia Campos.

We're soon going to have four public APIs:

 * GTK 3
 * GTK 4
 * WPE 1.0/1.1
 * WPE 2.0

To avoid having four different copies of all the public headers, let's
generate the public headers from a shared template. We'll start by
unifying the existing WPE and GTK headers without making any API
changes. Using a customized version of unifdef, we can use
PLATFORM(GTK) and PLATFORM(WPE) guards in the header templates in order
to hide platform-specific code from the generated headers. Be careful
around comments, though: the guards do not work inside comments, so if
any part of a doc comment needs to differ, the entire doc comment will
need to be duplicated.

Also note that I don't touch the DOM API. GTK 3 will just have to be
different from the other APIs.

Thanks to Adrian for proposing this solution.

This commit additionally removes the prohibition on tab characters under
Source/ThirdParty, since there's not really any great reason to care
about whitespace in third-party code, and we don't use Subversion
anymore so concerns about the Subversion allow-tabs property no longer
apply.

I also got rid of WebKitForwardDeclarations.h. Turns out it was only
needed to break an include cycle between GTK's WebKitWebView.h and
WebKitPrintOperation.h, but that can be solved easily enough without it.

I also added WebKitVersion.h to WPE's web process API, to match GTK.

* Source/ThirdParty/unifdef/CMakeLists.txt: Added.
* Source/ThirdParty/unifdef/README.webkit: Added.
* Source/ThirdParty/unifdef/unifdef.c: Added.
(processinout):
(cleantemp):
(version):
(synopsis):
(usage):
(help):
(Eelif):
(Eelse):
(Eendif):
(Eeof):
(Eioccc):
(print):
(drop):
(Strue):
(Sfalse):
(Selse):
(Pelif):
(Pelse):
(Pendif):
(Dfalse):
(Delif):
(Delse):
(Dendif):
(Fdrop):
(Fpass):
(Ftrue):
(Ffalse):
(Oiffy):
(Oif):
(Oelif):
(Idrop):
(Itrue):
(Ifalse):
(Mpass):
(Mtrue):
(Melif):
(Melse):
(ignoreon):
(keywordedit):
(nest):
(unnest):
(state):
(done):
(flushline):
(hashline):
(closeio):
(process):
(parseline):
(op_strict):
(op_lt):
(op_gt):
(op_le):
(op_ge):
(op_eq):
(op_ne):
(op_or):
(op_and):
(op_blsh):
(op_brsh):
(op_add):
(op_sub):
(op_mul):
(op_div):
(op_mod):
(op_bor):
(op_bxor):
(op_band):
(eval_unary):
(eval_table):
(ifeval):
(skiphash):
(skipline):
(skipcomment):
(skipargs):
(skipsym):
(getsym):
(matchsym):
(findsym):
(indirectsym):
(addsym1):
(addsym2):
(debugsym):
(defundefile):
(defundef):
(astrcat):
(xstrdup):
(debug):
(error):
* Source/ThirdParty/unifdef/unifdef.h: Added.
(mktempmode):
* Source/ThirdParty/unifdef/version.h: Added.
* Source/WebKit/GLib.cmake: Added.
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/UIProcess/API/glib/WebKitApplicationInfo.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitApplicationInfo.h.
* Source/WebKit/UIProcess/API/glib/WebKitAuthenticationRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitAuthenticationRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitAutocleanups.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitAutocleanups.h.
* Source/WebKit/UIProcess/API/glib/WebKitAutomationSession.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitAutomationSession.h.
* Source/WebKit/UIProcess/API/glib/WebKitBackForwardList.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitBackForwardList.h.
* Source/WebKit/UIProcess/API/glib/WebKitBackForwardListItem.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitBackForwardListItem.h.
* Source/WebKit/UIProcess/API/glib/WebKitContextMenu.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitContextMenu.h.
* Source/WebKit/UIProcess/API/glib/WebKitContextMenuActions.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitContextMenuActions.h.
* Source/WebKit/UIProcess/API/glib/WebKitContextMenuItem.h.in: Renamed from Source/WebKit/UIProcess/API/gtk3/WebKitContextMenuItem.h.
* Source/WebKit/UIProcess/API/glib/WebKitCookieManager.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitCookieManager.h.
* Source/WebKit/UIProcess/API/glib/WebKitCredential.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitCredential.h.
* Source/WebKit/UIProcess/API/glib/WebKitDefines.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitDefines.h.
* Source/WebKit/UIProcess/API/glib/WebKitDeviceInfoPermissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitDeviceInfoPermissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitDownload.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitDownload.h.
* Source/WebKit/UIProcess/API/glib/WebKitEditingCommands.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitEditingCommands.h.
* Source/WebKit/UIProcess/API/glib/WebKitEditorState.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitEditorState.h.
* Source/WebKit/UIProcess/API/glib/WebKitError.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitError.h.
* Source/WebKit/UIProcess/API/glib/WebKitFaviconDatabase.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitFaviconDatabase.h.
* Source/WebKit/UIProcess/API/glib/WebKitFileChooserRequest.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitFileChooserRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitFindController.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitFindController.h.
* Source/WebKit/UIProcess/API/glib/WebKitFormSubmissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitFormSubmissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitGeolocationManager.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitGeolocationManager.h.
* Source/WebKit/UIProcess/API/glib/WebKitGeolocationPermissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitGeolocationPermissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitHitTestResult.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitHitTestResult.h.
* Source/WebKit/UIProcess/API/glib/WebKitInputMethodContext.h.in: Renamed from Source/WebKit/UIProcess/API/gtk3/WebKitInputMethodContext.h.
* Source/WebKit/UIProcess/API/glib/WebKitInstallMissingMediaPluginsPermissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitInstallMissingMediaPluginsPermissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitJavascriptResult.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitJavascriptResult.h.
* Source/WebKit/UIProcess/API/glib/WebKitMediaKeySystemPermissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitMediaKeySystemPermissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitMemoryPressureSettings.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitMemoryPressureSettings.h.
* Source/WebKit/UIProcess/API/glib/WebKitMimeInfo.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitMimeInfo.h.
* Source/WebKit/UIProcess/API/glib/WebKitNavigationAction.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitNavigationAction.h.
* Source/WebKit/UIProcess/API/glib/WebKitNavigationPolicyDecision.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitNavigationPolicyDecision.h.
* Source/WebKit/UIProcess/API/glib/WebKitNetworkProxySettings.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitNetworkProxySettings.h.
* Source/WebKit/UIProcess/API/glib/WebKitNotification.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitNotification.h.
* Source/WebKit/UIProcess/API/glib/WebKitNotificationPermissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitNotificationPermissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitOptionMenu.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitOptionMenu.h.
* Source/WebKit/UIProcess/API/glib/WebKitOptionMenuItem.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitOptionMenuItem.h.
* Source/WebKit/UIProcess/API/glib/WebKitPermissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitPermissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitPlugin.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitPlugin.h.
* Source/WebKit/UIProcess/API/glib/WebKitPolicyDecision.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitPolicyDecision.h.
* Source/WebKit/UIProcess/API/glib/WebKitResponsePolicyDecision.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitResponsePolicyDecision.h.
* Source/WebKit/UIProcess/API/glib/WebKitScriptDialog.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitScriptDialog.h.
* Source/WebKit/UIProcess/API/glib/WebKitSecurityManager.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitSecurityManager.h.
* Source/WebKit/UIProcess/API/glib/WebKitSecurityOrigin.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitSecurityOrigin.h.
* Source/WebKit/UIProcess/API/glib/WebKitSettings.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitSettings.h.
* Source/WebKit/UIProcess/API/glib/WebKitURIRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitURIRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitURIResponse.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitURIResponse.h.
* Source/WebKit/UIProcess/API/glib/WebKitURISchemeRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitURISchemeRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitURISchemeResponse.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitURISchemeResponse.h.
* Source/WebKit/UIProcess/API/glib/WebKitURIUtilities.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitURIUtilities.h.
* Source/WebKit/UIProcess/API/glib/WebKitUserContent.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitUserContent.h.
* Source/WebKit/UIProcess/API/glib/WebKitUserContentFilterStore.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitUserContentFilterStore.h.
* Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitUserContentManager.h.
* Source/WebKit/UIProcess/API/glib/WebKitUserMediaPermissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitUserMediaPermissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitUserMessage.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitUserMessage.h.
* Source/WebKit/UIProcess/API/glib/WebKitVersion.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitVersion.h.in.
* Source/WebKit/UIProcess/API/glib/WebKitWebContext.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitWebContext.h.
* Source/WebKit/UIProcess/API/glib/WebKitWebResource.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitWebResource.h.
* Source/WebKit/UIProcess/API/glib/WebKitWebView.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitWebView.h.
* Source/WebKit/UIProcess/API/glib/WebKitWebViewSessionState.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitWebViewSessionState.h.
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteData.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitWebsiteData.h.
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataAccessPermissionRequest.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitWebsiteDataAccessPermissionRequest.h.
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.h.in: Renamed from Source/WebKit/UIProcess/API/wpe/WebKitWebsiteDataManager.h.
* Source/WebKit/UIProcess/API/glib/WebKitWebsitePolicies.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitWebsitePolicies.h.
* Source/WebKit/UIProcess/API/glib/WebKitWindowProperties.h.in: Renamed from Source/WebKit/UIProcess/API/gtk/WebKitWindowProperties.h.
* Source/WebKit/UIProcess/API/gtk/WebKitApplicationInfo.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitAuthenticationRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitAutomationSession.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitBackForwardList.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitBackForwardListItem.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitContextMenu.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitDefines.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitDeviceInfoPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitDownload.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitEditingCommands.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitEditorState.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitFindController.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitForwardDeclarations.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitGeolocationManager.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitGeolocationPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitHitTestResult.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitInstallMissingMediaPluginsPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitMediaKeySystemPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitMemoryPressureSettings.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitMimeInfo.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitNavigationAction.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitNetworkProxySettings.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitNotification.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitNotificationPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitOptionMenu.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitOptionMenuItem.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitPlugin.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitPolicyDecision.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.h:
* Source/WebKit/UIProcess/API/gtk/WebKitResponsePolicyDecision.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitScriptDialog.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitSecurityManager.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitSecurityOrigin.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitURIRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitURIResponse.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitURISchemeRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitURISchemeResponse.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitURIUtilities.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitUserContent.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitUserContentFilterStore.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitUserContentManager.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitUserMediaPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitUserMessage.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitVersion.h.in: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitWebResource.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.h.in: Renamed from Source/WebKit/UIProcess/API/gtk3/WebKitWebViewBase.h.
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewSessionState.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitWebsiteData.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitWebsiteDataAccessPermissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/gtk/WebKitWebsiteDataManager.h: Removed.
* Source/WebKit/UIProcess/API/gtk4/WebKitContextMenuItem.h: Removed.
* Source/WebKit/UIProcess/API/gtk4/WebKitInputMethodContext.h: Removed.
* Source/WebKit/UIProcess/API/gtk4/WebKitWebViewBase.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitAutocleanups.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitContextMenuActions.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitContextMenuItem.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitCookieManager.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitCredential.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitError.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitFaviconDatabase.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitFileChooserRequest.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitFormSubmissionRequest.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitInputMethodContext.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitJavascriptResult.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitNavigationPolicyDecision.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitSettings.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitWebContext.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitWebView.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitWebsitePolicies.h: Removed.
* Source/WebKit/UIProcess/API/wpe/WebKitWindowProperties.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitConsoleMessage.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitConsoleMessage.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitFrame.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitFrame.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitScriptWorld.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitScriptWorld.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebEditor.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebEditor.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebExtension.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebExtension.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebExtensionAutocleanups.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebExtensionAutocleanups.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebHitTestResult.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebHitTestResult.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebPage.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebPage.h.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/webkit-web-extension.h.in: Renamed from Source/WebKit/WebProcess/InjectedBundle/API/gtk/webkit-web-extension.h.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitConsoleMessage.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitScriptWorld.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebEditor.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebExtension.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebExtensionAutocleanups.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/WebKitWebHitTestResult.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitFrame.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/WebKitWebPage.h: Removed.
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/webkit-web-extension.h: Removed.
* Tools/Scripts/webkitpy/style/checker.py:

Canonical link: https://commits.webkit.org/253377@main
